Failing compilation on RP2040

Hello! Im sifting through all the various 5110 libraries and ive gotten yours closest to working! I think im nearly there, but I'm still getting some strange and currious compiliation issues:

```
❯ make flash
pio run -t upload
Processing pico (board: pico; platform: raspberrypi; framework: arduino)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/raspberrypi/pico.html
PLATFORM: Raspberry Pi RP2040 (1.7.0) > Raspberry Pi Pico
HARDWARE: RP2040 133MHz, 264KB RAM, 2MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, jlink, raspberrypi-swd)
PACKAGES: 
 - framework-arduino-mbed @ 3.1.1 
 - tool-openocd-raspberrypi @ 2.1100.0 (11.0) 
 - tool-rp2040tools @ 1.0.2 
 - toolchain-gccarmnoneeabi @ 1.90201.191206 (9.2.1)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 38 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Nokia 5110 LCD library @ 2.6.1
|   |-- SPI
Building in release mode
Compiling .pio/build/pico/src/main.cpp.o
Generating LD script .pio/build/pico/cpp.linker_script.ld
Compiling .pio/build/pico/lib40e/SPI/SPI.cpp.o
Compiling .pio/build/pico/libf35/Nokia 5110 LCD library/Nokia_LCD.cpp.o
Compiling .pio/build/pico/FrameworkArduinoVariant/double_tap_usb_boot.cpp.o
Compiling .pio/build/pico/FrameworkArduinoVariant/variant.cpp.o
Compiling .pio/build/pico/FrameworkArduino/Interrupts.cpp.o
Compiling .pio/build/pico/FrameworkArduino/Serial.cpp.o
Compiling .pio/build/pico/FrameworkArduino/Tone.cpp.o
In file included from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/api/String.h:33,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/api/IPAddress.h:24,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/api/ArduinoAPI.h:30,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/Arduino.h:27,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/libraries/SPI/SPI.h:21,
                 from .pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp:8:
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/api/deprecated-avr-comp/avr/pgmspace.h:108: warning: "pgm_read_byte_near" redefined
  108 | #define pgm_read_byte_near(addr) pgm_read_byte(addr)
      | 
.pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp:6: note: this is the location of the previous definition
    6 | #define pgm_read_byte_near *
      | 
Compiling .pio/build/pico/FrameworkArduino/USB/PluggableUSBDevice.cpp.o
Compiling .pio/build/pico/FrameworkArduino/USB/USBCDC.cpp.o
Compiling .pio/build/pico/FrameworkArduino/USB/USBSerial.cpp.o
Compiling .pio/build/pico/FrameworkArduino/WMath.cpp.o
Compiling .pio/build/pico/FrameworkArduino/abi.cpp.o
Compiling .pio/build/pico/FrameworkArduino/api/Common.cpp.o
Compiling .pio/build/pico/FrameworkArduino/api/IPAddress.cpp.o
Compiling .pio/build/pico/FrameworkArduino/api/PluggableUSB.cpp.o
Compiling .pio/build/pico/FrameworkArduino/api/Print.cpp.o
Compiling .pio/build/pico/FrameworkArduino/api/Stream.cpp.o
Compiling .pio/build/pico/FrameworkArduino/api/String.cpp.o
Compiling .pio/build/pico/FrameworkArduino/arm_hal_random.c.o
Compiling .pio/build/pico/FrameworkArduino/as_mbed_library/variant.cpp.o
In file included from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_regs/include/hardware/platform_defs.h:12,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2_common/pico_platform/include/pico/platform.h:12,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/common/pico_base/include/pico.h:19,
                 from /home/joe/.platformio/packages/framework-arduino-mbed/variants/RASPBERRY_PI_PICO/double_tap_usb_boot.cpp:2:
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2_common/hardware_timer/include/hardware/timer.h: In function 'uint32_t time_us_32()':
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_regs/include/hardware/regs/addressmap.h:56:20: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
   56 | #define TIMER_BASE 0x40054000
      |                    ^~~~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_structs/include/hardware/structs/timer.h:33:38: note: in expansion of macro 'TIMER_BASE'
   33 | #define timer_hw ((timer_hw_t *const)TIMER_BASE)
      |                                      ^~~~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2_common/hardware_timer/include/hardware/timer.h:66:12: note: in expansion of macro 'timer_hw'
   66 |     return timer_hw->timerawl;
      |            ^~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2_common/hardware_timer/include/hardware/timer.h: In function 'bool time_reached(absolute_time_t)':
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_regs/include/hardware/regs/addressmap.h:56:20: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
   56 | #define TIMER_BASE 0x40054000
      |                    ^~~~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_structs/include/hardware/structs/timer.h:33:38: note: in expansion of macro 'TIMER_BASE'
   33 | #define timer_hw ((timer_hw_t *const)TIMER_BASE)
      |                                      ^~~~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2_common/hardware_timer/include/hardware/timer.h:110:19: note: in expansion of macro 'timer_hw'
  110 |     uint32_t hi = timer_hw->timerawh;
      |                   ^~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_regs/include/hardware/regs/addressmap.h:56:20: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
   56 | #define TIMER_BASE 0x40054000
      |                    ^~~~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2040/hardware_structs/include/hardware/structs/timer.h:33:38: note: in expansion of macro 'TIMER_BASE'
   33 | #define timer_hw ((timer_hw_t *const)TIMER_BASE)
      |                                      ^~~~~~~~~~
/home/joe/.platformio/packages/framework-arduino-mbed/cores/arduino/mbed/targets/TARGET_RASPBERRYPI/TARGET_RP2040/pico-sdk/rp2_common/hardware_timer/include/hardware/timer.h:111:33: note: in expansion of macro 'timer_hw'
  111 |     return (hi >= hi_target && (timer_hw->timerawl >= (uint32_t) target || hi != hi_target));
      |                                 ^~~~~~~~
Compiling .pio/build/pico/FrameworkArduino/itoa.c.o
Compiling .pio/build/pico/FrameworkArduino/main.cpp.o
Compiling .pio/build/pico/FrameworkArduino/mbed/platform/cxxsupport/mstd_mutex.cpp.o
Compiling .pio/build/pico/FrameworkArduino/pinToIndex.cpp.o
Compiling .pio/build/pico/FrameworkArduino/random_seed.cpp.o
Compiling .pio/build/pico/FrameworkArduino/timer.cpp.o
.pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp: In member function 'bool Nokia_LCD::send(unsigned char, bool, bool)':
.pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp:293:13: error: 'class arduino::MbedSPI' has no member named 'setClockDivider'
  293 |         SPI.setClockDivider(SPI_CLOCK_DIV4);
      |             ^~~~~~~~~~~~~~~
.pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp:293:29: error: 'SPI_CLOCK_DIV4' was not declared in this scope
  293 |         SPI.setClockDivider(SPI_CLOCK_DIV4);
      |                             ^~~~~~~~~~~~~~
.pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp:294:13: error: 'class arduino::MbedSPI' has no member named 'setDataMode'
  294 |         SPI.setDataMode(SPI_MODE0);
      |             ^~~~~~~~~~~
.pio/libdeps/pico/Nokia 5110 LCD library/src/Nokia_LCD.cpp:295:13: error: 'class arduino::MbedSPI' has no member named 'setBitOrder'
  295 |         SPI.setBitOrder(MSBFIRST);
      |             ^~~~~~~~~~~
Compiling .pio/build/pico/FrameworkArduino/wiring.cpp.o
*** [.pio/build/pico/libf35/Nokia 5110 LCD library/Nokia_LCD.cpp.o] Error 1
============================================================================================ [FAILED] Took 1.39 seconds ============================================================================================
make: *** [Makefile:22: flash] Error 1
```

I looked around online and did not see anything else like this but, what i suspect is perhaps the implementation is slightly different for the arduino platform on another chip like the rp2040? Platformio says no, its all the same, but these methods just, straight up dont exist for me somehow? or maybe im missing a dependancy?

Heres my code and the pio file along with it if you want to take a look, just to reduce complexity i stripped it down a little bit. https://github.com/Kitsune-Robotics/kw1fox-2/blob/7b35d6ea773fcefeac901daae8ec5f9a656755b2/Firmware/src/main.cpp

I'll try to compile for another board with a samd or atmel processor to see if theres a difference, if there is, maybe its just broken on the rp2040? We can add that to the library.json to mark it as incompatable or. Maybe im just being dumb and i forgot to bump a version or, something. Feedback appreciated!

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=Nokia 5110 LCD library
-version=2.6.1
+version=2.6.2
 author=Dimitris Platis
 maintainer=Dimitris Platis <dimitris@plat.is>
 sentence=Arduino library for driving the Nokia 5110 LCD

--- a/src/Nokia_LCD.cpp
+++ b/src/Nokia_LCD.cpp
@@ -1,5 +1,5 @@
 // If we are not building for AVR architectures ignore PROGMEM
-#if defined(__AVR__)
+#if __has_include(<avr/pgmspace.h>)
 #include <avr/pgmspace.h>
 #else
 #define PROGMEM
@@ -98,6 +98,10 @@ void Nokia_LCD::begin() {
     // Reset the LCD to a known state
     digitalWrite(kRst_pin, LOW);
     digitalWrite(kRst_pin, HIGH);
+
+    if (kUsingHardwareSPI) {
+        SPI.begin();
+    }
 
     sendCommand(0x21);  // Tell LCD extended commands follow
     sendCommand(0xB0);  // Set LCD Vop (Contrast)
@@ -289,12 +293,10 @@ bool Nokia_LCD::send(const unsigned char lcd_byte, const bool is_data,
     // Send the byte
     digitalWrite(kCe_pin, LOW);
     if (kUsingHardwareSPI) {
-        SPI.begin();
-        SPI.setClockDivider(SPI_CLOCK_DIV4);
-        SPI.setDataMode(SPI_MODE0);
-        SPI.setBitOrder(MSBFIRST);
+        constexpr uint32_t kSPiClockSpeed{F_CPU / 4U};
+        SPI.beginTransaction(SPISettings{kSPiClockSpeed, MSBFIRST, SPI_MODE0});
         SPI.transfer(lcd_byte);
-        SPI.end();
+        SPI.endTransaction();
     } else {
         shiftOut(kDin_pin, kClk_pin, MSBFIRST, lcd_byte);
     }

